addons/website_slides_survey/tests/test_course_certification_stats.py

```diff
@@ -0,0 +1,125 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.addons.mail.tests.common import mail_new_test_user
+from odoo.tests.common import TransactionCase
+
+
+class TestCourseCertificationStats(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Create courses and certifications
+        cls.channels = cls.env["slide.channel"].create(
+            [
+                {
+                    "name": f"Course {i +1}",
+                    "enroll": "public",
+                    "is_published": True,
+                }
+                for i in range(3)
+            ]
+        )
+
+        cls.certifications = cls.env["survey.survey"].create(
+            [
+                {
+                    "title": f"certification {i + 1}",
+                    "scoring_type": "scoring_with_answers",
+                    "certification": True,
+                }
+                for i in range(3)
+            ]
+        )
+
+        # Link the certifications to courses
+        cls.slide_certifications = cls.env["slide.slide"].create(
+            [
+                {
+                    "name": f"Certification {i + 1} slide",
+                    "channel_id": cls.channels.ids[i],
+                    "slide_category": "certification",
+                    "survey_id": cls.certifications.ids[i],
+                    "is_published": True,
+                }
+                for i in range(3)
+            ]
+        )
+
+        # Create course participants
+        cls.portal_users = [
+            mail_new_test_user(
+                cls.env,
+                login=f"portal_test {i + 1}",
+                groups="base.group_portal",
+                name=f"Portal {i + 1}",
+            )
+            for i in range(3)
+        ]
+
+        cls.channel_partners = cls.env["slide.channel.partner"].create(
+            [
+                {
+                    "channel_id": channel_id,
+                    "partner_id": user.partner_id.id,
+                }
+                for channel_id in cls.channels.ids
+                for user in cls.portal_users
+            ]
+        )
+
+        # Set certified/not certified participants
+        cls.slide_partners = cls.env["slide.slide.partner"].create(
+            [
+                {
+                    "channel_id": channel_id.id,
+                    "partner_id": user.partner_id.id,
+                    "slide_id": channel_id.slide_ids.ids[0],
+                }
+                for channel_id in cls.channels
+                for user in cls.portal_users
+            ]
+        )
+
+        cls.slide_partners[0].survey_scoring_success = True
+        cls.slide_partners[-1].survey_scoring_success = True
+
+    def test_channel_members_certified_count(self):
+        members_certified_counts = self.channels.mapped("members_certified_count")
+        self.assertEqual(
+            members_certified_counts[0],
```

Comment:
Would assert one by one based on cls.channel_1, cls.channel_2 etc, not a fan of index-based access.